http-clients/netty-nio-client/src/main/java/software/amazon/awssdk/http/nio/netty/internal/ChannelPipelineInitializer.java

```diff
@@ -99,7 +99,9 @@ public void channelCreated(Channel ch) {
         ch.attr(CHANNEL_DIAGNOSTICS).set(new ChannelDiagnostics(ch));
         ch.attr(PROTOCOL_FUTURE).set(new CompletableFuture<>());
         ChannelPipeline pipeline = ch.pipeline();
-        if (sslCtx != null) {
+
+        boolean sslCtxPresent = sslCtx != null;
```

Comment:
whats the reason for this change ?

Reply:
We are saving the boolean as variable since we pass to `configureProtocolHandlers()` to fallback to prior knowledge if sslCtx is null

Reply:
This is existing behavior, to use prior knowledge. Covered by test here, to make sure we don't throw `UnsupportedOperationException`
https://github.com/aws/aws-sdk-java-v2/blob/ed58871b0bc410944537c433a2b02346fea111a0/test/codegen-generated-classes-test/src/test/java/software/amazon/awssdk/services/h2/AlpnHttpTest.java#L49